File: TempRecycle/Utils/Remove file.cs
using System.Collections.Concurrent;

namespace TempRecycle.Utils
{
    public class Remove_file
    {
        public static async Task RemoveALl(string path, int folderCount, int filesCount)
        {
            int currentLine = Console.CursorTop;

            Console.WriteLine("\nDo you want to delete the scanned files? (Y/N):");
            string? respuesta = Console.ReadLine()?.Trim().ToUpper();


            if (string.IsNullOrEmpty(respuesta))
            {
                BoxMessage.ShowError("No data was written", ref currentLine);
                return;
            }

            var files = new ConcurrentBag<string>(Directory.EnumerateFiles(path, "*", SearchOption.AllDirectories).Where(f => !f.Contains("Microsoft.CodeAnalysis") && !f.Contains("Windows")));


            var directories = new ConcurrentBag<string>(Directory.EnumerateDirectories(path, "*", SearchOption.AllDirectories));


            Console.WriteLine();

            if (respuesta == "Y")
            {
                foreach (var file in files)
                {
                    try
                    {
                        if (File.Exists(file))
                        {
                            File.SetAttributes(file, FileAttributes.Normal);
                            File.Delete(file);
                            BoxMessage.ShowInfo("Deleted Files", ref currentLine);
                        }

                    }
                    catch (UnauthorizedAccessException ex)
                    {
                        BoxMessage.ShowError($"Could not in {file} : {ex.Message}", ref currentLine);
                    }
                    catch (IOException ex)
                    {
                        BoxMessage.ShowError($"Could not in {file} : {ex.Message}",  ref currentLine);
                    }
                    await Task.Yield();
                }
                //folders
                foreach (var dir in directories)
                {
                    try
                    {
                        Directory.Delete(dir);
                        BoxMessage.ShowInfo("Deleted Folders", ref currentLine);
                    }
                    catch (UnauthorizedAccessException ex)
                    {
                        BoxMessage.ShowError($"Could not in {dir} : {ex.Message}", ref currentLine);
                    }
                    catch (IOException ex)
                    {
                        BoxMessage.ShowError($"Could not in {dir} : {ex.Message}", ref currentLine);
                    }
                }
            }
            else if (respuesta == "N")
            {
                BoxMessage.ShowInfo("Cancel Operation", ref currentLine);
            }
            else
            {
                BoxMessage.ShowError("Faild Operation", ref currentLine);
            }
        }
    }

}

File: utils/RemoveFile.cpp
#include "RemoveFile.h"
#include "BoxMessage.h"
#include <filesystem>
#include <windows.h>
#include <iostream>

namespace fs = std::filesystem;

namespace RemoveFile {
    void RemoveAll(const std::string& path, int fileCount, int folderCount) {
        std::string input;
        std::cout << "\nDo you want to delete the scanned files? (Y/N): ";
        std::getline(std::cin, input);

        if (input.empty()) {
            BoxMessage::ShowError("No data was written");
            return;
        }

        if (input != "Y" && input != "y") {
            BoxMessage::ShowInfo("Cancel Operation");
            return;
        }

        // Delete files
        for (auto& entry : fs::recursive_directory_iterator(path,
                 fs::directory_options::skip_permission_denied)) {
            try {
                if (entry.is_regular_file()) {
                    std::string pathStr = entry.path().string();
                    DWORD attr = GetFileAttributesW(entry.path().c_str());

                    if (attr != INVALID_FILE_ATTRIBUTES &&
                        (attr & (FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM))) {
                        continue;
                    }

                    fs::remove(entry.path());
                    BoxMessage::ShowInfo("Deleted file: " + pathStr);
                }
            } catch (const std::exception& ex) {
                BoxMessage::ShowError("Could not delete file: " + std::string(ex.what()));
            }
        }

        // Delete empty folders
        for (auto it = fs::recursive_directory_iterator(path,
                     fs::directory_options::skip_permission_denied);
             it != fs::recursive_directory_iterator(); ++it) {
            try {
                if (it->is_directory() && fs::is_empty(*it)) {
                    fs::remove(*it);
                    BoxMessage::ShowInfo("Deleted folder: " + it->path().string());
                }
            } catch (const std::exception& ex) {
                BoxMessage::ShowError("Could not delete folder: " + std::string(ex.what()));
            }
        }

        BoxMessage::ShowInfo("Cleanup completed.");
    }
}
